README.md

```diff
@@ -132,3 +132,97 @@ method: http
 http:
    port: 3903
 ```
+
+## TLS configuration
+
+You can use exporter_exporter with TLS to encrypt the traffic, and at the
+same time enforce strong mutual authentication between the nodes and the
+prometheus server.
+
+Note that `-web.tls.verify` will accept *any* certificate signed by the
+`-web.tls.ca`, so you need to create a separate CA for this purpose - or use
+a self-signed certificate, which acts as its own CA.
+
+Here is a simple configuration example, using one key/cert for the
+prometheus server and one key/cert shared between all the remote nodes.
+Firstly, create the keys and certs:
+
+```
+openssl req -x509 -newkey rsa:1024 -keyout prom_node_key.pem -out prom_node_cert.pem -days 29220 -nodes -subj /commonName=prom_node/
```

Comment:
I did consider using a proper CA: initially I was doing to show a single CA signing two certificates, one for node and one for prometheus.

However I don't see value in doing so unless exporter_exporter has a way to validate the client certificate presented to it.  (e.g. could add new options `-web.tls.accept.cn=prometheus` or `-web.tls.accept.fingerprint=xxxxxxxx`)

Without this, exporter_exporter accepts *any* certificate, and in particular, one node could use its own key and certificate to scrape any other node.  This forces you to have two CAs.  But if you're going to do that, you might as well keep it simple and use self-signed certs.

Reply:
Well, it accepts any cert authorized by the signing CA. This makes it easier to rotate the cert used by the caller (vs only allowing one specific cert). In my experience, this is the best approach for managable mTLS.

Reply:
With respect, certificates don't "authorize", they only "identify".  You can issue certs to all the people and servers in your organization from your organization's CA; this doesn't mean you give them permission to do anything they like.

The specific problem here is: we need to issue a cert to the node, so that prometheus can confirm it's talking securely to the node (without any MITM in progress) - that is, to confirm the identity of the node.

From the node's point of view, it needs to accept incoming scrape connections from the prometheus server, and similarly confirm its identity to ensure no MITM; but in addition, it should only allow scrapes from the prometheus server, not from other nodes, or other holders of certs from the same CA for that matter.

The standard way to deal with this is to perform a separate authorization process based on the identity.  In this case, the node would check that the identity of the client is "cn=prometheus" (say), and only accept if that is true.

This still allows keys to be rotated without problems: if the prometheus node needs a new key for itself, then the CA issues a new cert with identity "cn=prometheus", and optionally revokes the old one.

If you want to take the simple approach and authorize everyone who holds a cert from the CA, then you need a separate CA for each authorization purpose.

(Aside: there *are* a few authorization flags you can put in a cert, such as CA:true or server/client usage constraints. They're not sufficient for the purpose here, although you could have a certificate check which says "I authorize everyone who holds a client cert, but not a server cert")

Reply:
Sorry, hadn't noticed this had been merged anyway :-)

Reply:
My point re rotation was mostly a response to your comment about fingerprints. For the purposes of authz/authn, multiple CAs work (thing issues by a monitoring agents CA can do the monitoring things). I'd definitely accept a PR to match cert attributes in some way (thought I think cn specifically is no longer considered best practice)

Reply:
Accepted: fingerprints are definitely not best practice here.  For client certs, I'm not sure what you'd use in place of cn (emailAddress?)

Anyway, I have this deployed now with self-signed certs, so implementing VerifyPeerCertificate will be a back-burner project.  Thanks for accepting the PR, it means I don't have to keep separate documentation :-)